### PR TITLE
[optim][muon] fix momentum buffer aliasing in Newton-Schulz

### PR DESCRIPTION
**Pull request by weifengpy**
Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.14.0) (oldest at bottom):
* #190597
* __->__ #194664

`_zeropower_via_newtonschulz` happens in-place. 
* before: `grad.bfloat16()` returns itself when already bfloat16, and then normalized it in place
via `div_`
* after: use `grad.to(dtype=torch.bfloat16, copy=True)` so the input is never aliased

### CODE REVIEW

File: test/test_optim.py [hunk 1]
```diff
@@ -2701,8 +2701,32 @@ def fwd_bwd(optim, mod, i):
         self.assertEqual(counter, 6)
 
 
+class TestMuon(TestCase):
+    def test_muon_does_not_alias_momentum_buffer(self, device):
```
Comment (janeyx99) [inline test/test_optim.py]: move the test to TestOptimRenewed, delete the long unreadable comment: 

```
      @dtypes(torch.float32, torch.bfloat16)
      @parametrize("nesterov", [False, True])
      def test_muon_orthogonalization_does_not_alias_momentum_buffer(self, device, dtype, nesterov):
          momentum = 0.9
          param = Parameter(torch.zeros(8, 5, device=device, dtype=dtype))
          param.grad = torch.ones_like(param)
          optimizer = torch.optim.Muon(
              [param],
              lr=0,
              weight_decay=0,
              momentum=momentum,
              nesterov=nesterov,
          )
          expected = torch.zeros_like(param)
          expected.lerp_(param.grad, 1 - momentum)
          optimizer.step()
          self.assertEqual(
              optimizer.state[param]["momentum_buffer"],
              expected,
          )
```

Comment (weifengpy) [reply test/test_optim.py]: done

Comment (weifengpy) [reply test/test_optim.py]: thanks for the concrete example

**Review by janeyx99 — APPROVED** [review]
